app/Models/Mship/Account.php

```diff
@@ -512,4 +514,35 @@ public function __toString()
     {
         return $this->name;
     }
+
+    /**
+     * Return a valid access token for the user or return null if none.
+     *
+     * @return \League\OAuth2\Client\Token\AccessToken
+     * @return null
+     */
+    public function getTokenAttribute()
+    {
+        if ($this->vatsim_access_token === null) {
```

Comment:
Could you invert this loop so the bulk of the logic is in the if part and the else is the secondary condition? Also might be worth throwing an exception instead of returning null.

Reply:
Removed the entire method as right now we don't need it - we only use SSO for login and not for updating user info. If we do in the future, we can re-add this logic.

Same with Discord. We store an access and refresh token but do not use them.

ecb238f8